src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -222,8 +232,9 @@ def set_config(self, config: object):
         multicluster_config = self.load_multi_cluster_config()
         multicluster_config.update({'current_url': config['url']})
         multicluster_config.update({'current_user': config['user']})
-        Settings.MULTICLUSTER_CONFIG = multicluster_config
-        return Settings.MULTICLUSTER_CONFIG
+        Settings.MULTICLUSTER_CONFIG = json.dumps(multicluster_config)
+        multi_cluster_config = self.load_multi_cluster_config()
+        return multi_cluster_config
```

Comment:
Is there any reason to load it again? why not return the non jsonified item?
```suggestion
        return multicluster_config
```